inspectit-gepard-agent/src/main/java/rocks/inspectit/gepard/agent/instrumentation/hook/action/span/SpanAction.java

```diff
@@ -1,35 +1,47 @@
 /* (C) 2024 */
-package rocks.inspectit.gepard.agent.instrumentation.hook.action;
+package rocks.inspectit.gepard.agent.instrumentation.hook.action.span;
 
+import io.opentelemetry.api.common.Attributes;
 import io.opentelemetry.api.trace.Span;
 import io.opentelemetry.api.trace.Tracer;
 import io.opentelemetry.context.Context;
 import java.util.Objects;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import rocks.inspectit.gepard.agent.instrumentation.hook.action.exception.CouldNotCloseSpanScopeException;
-import rocks.inspectit.gepard.agent.instrumentation.hook.action.util.SpanUtil;
+import rocks.inspectit.gepard.agent.instrumentation.hook.action.MethodExecutionContext;
+import rocks.inspectit.gepard.agent.instrumentation.hook.action.span.exception.CouldNotCloseSpanScopeException;
+import rocks.inspectit.gepard.agent.instrumentation.hook.action.span.util.SpanUtil;
 import rocks.inspectit.gepard.agent.internal.otel.OpenTelemetryAccessor;
 
 /** This action contains the logic to start and end a {@link Span}. */
 public class SpanAction {
   private static final Logger log = LoggerFactory.getLogger(SpanAction.class);
 
   /**
-   * Starts a new {@link Span}. Should be called before {@link SpanAction#endSpan}.
+   * Starts a new {@link Span} and adds the provided attributes to the current span. If the
+   * currently active span already uses the provided spanName, no new span will be created. Should
+   * be called before {@link SpanAction#endSpan}.
    *
-   * @param spanName the name of the span
+   * @param executionContext the execution context of the current method
    * @return the scope of the started span or null, if the current span has the same name
    */
-  public AutoCloseable startSpan(String spanName) {
-    // In the future, we still might want to set some attributes in the current span
+  public AutoCloseable startSpan(MethodExecutionContext executionContext) {
+    String spanName = getSpanName(executionContext);
+    Attributes methodAttributes = SpanUtil.createMethodAttributes(executionContext);
+
+    // Use the OTel span
     if (SpanUtil.spanAlreadyExists(spanName)) {
       log.debug("Span '{}' already exists at the moment. No new span will be started", spanName);
+      Span otelSpan = Span.current();
+      // We overwrite the OTel attributes, if they use the same key
+      otelSpan.setAllAttributes(methodAttributes);
       return null;
```

Comment:
Wouldn´t it make sense to return `otelSpan().makeCurrent()`?

Reply:
Or maybe an optional and the name could be startSpanIfDoesntExist or something like that.

Reply:
OTel has already called `makeCurrent()` at that point. Otherwiese `Span.current()` would not return it.
We should also not return anything from the otelSpan, since OTel will do all the important work with it.

Reply:
Always forget to use Optional instead of null...